File: JavaScript/7js 4/validSpacing.js
/*
|--------------------------------------------------------------------------
| Valid Spacing
|--------------------------------------------------------------------------
|
*/

function validSpacing(s) {
   return s.split(" ").includes("") == false || s == "";
}


function validSpacing(s) {
   return s.replace(/\s+/g, " ").trim() == s;
}


function validSpacing(s) {
   return s.trim() == s && !s.includes("  ");
}


function validSpacing(s) {
   return !/^ |  | $/.test(s);
}


console.log(validSpacing("Hello world"), true);
console.log(validSpacing(" Hello world"), false);
console.log(validSpacing("Hello  world "), false);
console.log(validSpacing(""), true);

File: Python/7py 4/valid_spacing.py
"""
|--------------------------------------------------------------------------
| Valid Spacing
|--------------------------------------------------------------------------
|
"""

def valid_spacing(s):
   return '' not in s.split(' ') or s == ''


def valid_spacing(s):
   return s == ' '.join(s.split())


def valid_spacing(s):
   return s == s.strip() and '  ' not in s


print(valid_spacing('Hello world'),True)
print(valid_spacing(' Hello world'),False)
print(valid_spacing('Hello  world '),False)
print(valid_spacing(''),True)